### ISSUE TITLE
Can not call taskToggleCommand when the cursor is at the first or last position of a task

### ISSUE THREAD

**Issue by ekaragodin** [silverbulletmd/silverbullet#238]
It looks like the context in the first and last positions of the task is detected as "Document" instead of "Task".

**Comment by ekaragodin** [comment]
After investigating this bug I found there are several cases where [context detection by the current node](https://github.com/silverbulletmd/silverbullet/blob/cca48f66cdfe0a93a5f6d6ce32df609d66a1ab99/web/editor.tsx#L1076) works badly:

```
- [  ] My task
              ^ context = Document
- [  ] My task
    ^ context = TaskMarker
- [  ] My task
 ^ context = ListItem
- [  ] Meet with [[Jhon]]
                   ^ context = WikiLinkPage
```

I think they need a better way to detect task context.

**Comment by zefhemel** [comment]
Yeah, this was done in a very ad hoc manner for a very specific use case. I'll have a look at your PR a bit later. Thanks!

**Comment by janssen-io** [comment]
I tried running the `Task: Cycle State` command from various locations inside a task and they all seem to work now. :) Unless anyone else can still reproduce this, we can close this issue.
